_posts/yiyangshi/2016-04-29-YiyangsFinalProject.md

```diff
@@ -0,0 +1,121 @@
+
+3.Average delay minutes for different day of week
+
+This functionality is also achieved by the dictionary idea. Here are the two dictionaries:
+
+•	day_total_delay_minutes = dict(key = day of week, value = total delay minutes)
+
+•	day_delay_count = dict(key = day of week, value = number of delays)
+
+Day of week is represented by numeric number in the original dataset. So I created an extra dictionary to map the numbers to the strings so that the print out is more informative.
+
+4.Max, avg, min delay minutes for a(n) chosen Carrier or Airport
+
+This functionality allows users to pass in a carrier code or airport code and calculate the max, avg, min delay minutes for the carrier or airport. The calculations of max, avg, min are pretty standard so I would not elaborate here.
+
+Most of the times, user will not know the carrier code or airport code so I created help functions for users to search for the codes. For the carrier code, since there are only 12 carriers, I just printed out the whole reference list for user. Airport codes are too much to print, so for the airport code, I created a regex search function, which is similar to the Google location search homework we did. User can type in a blurry location and the program will return all the airports with the codes that matches the location user entered. User can then use the airport codes to find out the basic statistics.
+
+5.Explore Delay Reasons for a(n) chosen Carrier or Airport
+
+This is the most complicated piece of code. However, the basic idea is still the dictionary idea. This functionality is more like the combination of all the functionalities above. 
+
+For the specific input, two dictionaries are calculated:
+
+•	input_delay_minutes_by_delay_reason = dict(key = delay reason, value = total delay minutes)
+
+•	input_delay_count_by_delay_reason = dict(key = delay reason, value = number of delays)
+
+Similary, the calculation here are decided by the delay minutes of the specific delay reason. I also created a counter input_total_delay_count to keep track of the total number of delays caused by this specific input. Using the two dictionaries, we can get the average delay time breakdown by delay reasons. Using the second dictionary and the counter, we can get the percentage of delay caused by each delay reasons.
+
+Notice that, all of the calculation is based on the user input. Thus, the results are for a specific carrier or airport. This functionality also allows carrier code and airport code search.
+There are several new things that I learned from this project: the timer, the sort function, and the usefulness of dictionary. Honestly, I did not feel completely comfortable with list and dictionary. However, now I feel confident.
+
+Finally, a big thank you to you for this awesome class! I can now use Python to achieve so many things that I couldn’t before.
```

Comment:
Thank *you* for putting in a great effort all semester!  I plotted the course but you did the hard work of getting here 🚀